### PR TITLE
EJS: Fixed local override of async from true to false

### PR DESCRIPTION
**Pull request by TTPO100AJIEX**
There has been an error overriding async from true to false in the render function. These changes should fix that.
Now it is possible to specify async: true in global options, and async: false as a local option. The overriding behavior now works correctly: local option async: false will be used.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -376,7 +376,7 @@ function fastifyView (fastify, opts, next) {
         if (!this.getHeader('content-type')) {
           this.header('Content-Type', 'text/html; charset=' + charset)
         }
-        if (globalOptions.async || opts?.async) {
+        if (opts?.async !== undefined ? opts.async : globalOptions.async) {
```
Comment (climba03003) [inline index.js:379]:
```suggestion
        if (opts?.async ?? globalOptions.async) {
```

File: index.js [hunk 2]
```diff
@@ -279,7 +279,7 @@ function fastifyView (fastify, opts, next) {
       } catch (error) {
         cachedPage = error
       }
-      if (type === 'ejs' && (globalOptions.async || localOptions?.async)) {
+      if (type === 'ejs' && (localOptions.async)) {
```
Comment (climba03003) [inline index.js:282]:
```suggestion
      if (type === 'ejs' && (localOptions.async ?? globalOptions.async)) {
```

**Review by climba03003 — CHANGES_REQUESTED** [review]
You can use nullish operator and it can ensure the `local` take precedence when it has value.

**Review by climba03003 — APPROVED** [review]
LGTM.

**Review by lilpoozie2005 — APPROVED** [review]

**Pull request merged**